### PR TITLE
Track more asset metadata

### LINKED ISSUE #24: User content metadata tracking
- [x] Track who uploaded an asset, and when
- [x] Track type of asset (https://github.com/LittleBigRefresh/Refresh/issues/22)
- [ ] Track dependencies of asset
- [x] Add migration to backfill data into database from DataStore
- [ ] Track size of asset (https://github.com/LittleBigRefresh/Refresh/issues/66)

We can use this information to be able to purge unused assets/delete assets by a certain user

### LINKED ISSUE #66: Implement sizeOfResources on Levels

### PR DESCRIPTION
**Pull request by jvyden**
Closes #24 and closes #66

This PR implements support for parsing the dependency tree in binary game assets, and records the size of all assets.

### CODE REVIEW

File: Refresh.GameServer/Importing/AssetImporter.cs [hunk 1]
```diff
@@ -90,8 +89,77 @@ public void ImportFromDataStore(GameDatabaseContext context, IDataStore dataStor
...
+        ms.Seek(dependencyTableOffset, SeekOrigin.Begin);
+        uint dependencyCount = reader.ReadUInt32();
+
+        this.Debug($"Dependency count offset: {dependencyTableOffset}, count: {dependencyCount}");
+        for (int i = 0; i < dependencyCount; i++)
+        {
+            byte flags = reader.ReadByte();
+            if ((flags & 0x1) != 0) // UGC/SHA1
+            {
+                byte[] hashBytes = reader.ReadBytes(20);
```
Comment (Beyley) [inline Refresh.GameServer/Importing/AssetImporter.cs:155]: This allocates, there should be a stack allocation outside the `for` loop that you then read into using `.ReadAtLeast`

**Review by Beyley — CHANGES_REQUESTED** [review]
Only a small perf note, other than that it looks good

**Comment by Beyley** [comment]
side note: i wonder if this could be used to "garbage collect" some types of unreferenced assets

**Comment by jvyden** [comment]
Probably not worth chasing. Assets are small, best not to accidentally remove something that's in use when we're not expecting it.

**Review by Beyley — APPROVED** [review]

**Pull request merged**